packages/react/src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import IconButton from '../IconButton';
+import TooltipTabstop from '../TooltipTabstop';
+import Icon from '../Icon';
+
+type PaginationVariant = 'dark';
+
+interface Props {
+  totalItems: number;
+  itemsPerPage?: number;
+  currentPage?: number;
+  statusLabel?: React.ReactElement;
+  firstPageLabel?: string;
+  previousPageLabel?: string;
+  nextPageLabel?: string;
+  lastPageLabel?: string;
+  onNextPageClick?: () => void;
+  onPreviousPageClick?: () => void;
+  onFirstPageClick?: () => void;
+  onLastPageClick?: () => void;
+  className?: string;
+  variant?: PaginationVariant;
+}
+
+const Pagination = React.forwardRef<HTMLDivElement, Props>(
+  (
+    {
+      totalItems,
+      itemsPerPage = 10,
+      currentPage = 1,
+      statusLabel,
+      firstPageLabel = 'First page',
+      previousPageLabel = 'Previous page',
+      nextPageLabel = 'Next page',
+      lastPageLabel = 'Last page',
+      onNextPageClick,
+      onPreviousPageClick,
+      onFirstPageClick,
+      onLastPageClick,
+      className,
+      variant,
+      ...other
+    },
+    ref
+  ) => {
+    const itemStart = currentPage * itemsPerPage - itemsPerPage + 1;
+    const itemEnd = Math.min(itemStart + itemsPerPage - 1, totalItems);
+    const isLastPage = itemEnd === totalItems;
+
+    return (
+      <div
+        ref={ref}
+        className={classNames('Pagination', className, {
+          'Pagination--dark': variant === 'dark'
+        })}
+        {...other}
+      >
+        <ul>
+          <li>
+            {currentPage === 1 ? (
+              <TooltipTabstop tooltip={firstPageLabel} placement="bottom">
+                <Icon type="chevron-double-left" />
+              </TooltipTabstop>
+            ) : (
+              <IconButton
+                icon="chevron-double-left"
+                tooltipPlacement="bottom"
+                label={firstPageLabel}
+                onClick={onFirstPageClick}
+              />
+            )}
+          </li>
+
+          <li>
+            {currentPage === 1 ? (
+              <TooltipTabstop tooltip={previousPageLabel} placement="bottom">
+                <Icon type="chevron-left" />
+              </TooltipTabstop>
+            ) : (
+              <IconButton
+                icon="chevron-left"
+                tooltipPlacement="bottom"
+                label={previousPageLabel}
+                onClick={onPreviousPageClick}
+              />
```

Comment:
Some thoughts here (not anything we need to change for this PR):

We may want to consider eventually adding a custom render function here because this is making an assumption that pagination is going to be working in state, but it's possible they could be navigation based pagination. It doesn't seem like it's anything we need _right now_, but something to keep in mind for the future.